### PR TITLE
fix: update validation logic for numerical questions

### PR DESCRIPTION
**Pull request by sjschlapbach**

### CODE REVIEW

**Review by sjschlapbach — COMMENTED** [review]
@rschlaefli If we want to ensure also on the backend side that only responses in the specified range can be submitted, the `backend-response-processor` probably also needs a few more lines of code. Until now, this is only ensured for learning elements and micro sessions through the corresponding mutation that is called there.

File: packages/graphql/src/services/learningElements.ts [hunk 1]
```diff
@@ -219,9 +219,14 @@ export async function respondToQuestionInstance(
       }
 
       case QuestionType.NUMERICAL: {
-        if (!response.value) {
-          break
+        if (
+          !response.value ||
```
Comment (rschlaefli) [inline packages/graphql/src/services/learningElements.ts]:
```suggestion
          typeof response.value === 'undefined' ||
```

File: packages/shared-components/src/StudentQuestion.tsx [hunk 2]
```diff
@@ -129,15 +129,20 @@ export const StudentQuestion = ({
     }
   }
 
-  const onNumericalValueChange = (inputValue: any): void => {
-    const inputEmpty =
-      inputValue !== 0 && (!inputValue || inputValue.length === 0)
+  const onNumericalValueChange = (inputValue: string): void => {
+    const inputEmpty = !inputValue || inputValue.length === 0
```
Comment (rschlaefli) [inline packages/shared-components/src/StudentQuestion.tsx]:
```suggestion
    const inputEmpty = typeof inputValue === 'undefined' || inputValue === ''
```

**Review by rschlaefli — CHANGES_REQUESTED** [review]

File: packages/graphql/src/services/learningElements.ts (hunk 1 above)
Comment (sjschlapbach) [reply packages/graphql/src/services/learningElements.ts]: Shouldn't be an issue for strings, but I still adjusted it to test for undefined values, empty strings and null values as well (just to be sure 😄)

File: packages/shared-components/src/StudentQuestion.tsx (hunk 2 above)
Comment (sjschlapbach) [reply packages/shared-components/src/StudentQuestion.tsx]: Same as above

**Review by rschlaefli — APPROVED** [review]

**Pull request merged**